Abhi-AnimeX/app/src/main/java/com/abhijith/animex/di/RepositoryModule.kt

```diff
@@ -0,0 +1,21 @@
+package com.abhijith.animex.di
+
+import com.abhijith.animex.data.network.AnimeXApiService
+import com.abhijith.animex.data.repository.ISeasonalAnimeListRepositoryImpl
+import com.abhijith.animex.domain.repository.ISeasonalAnimeListRepository
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object RepositoryModule {
+
+    @Singleton
+    @Provides
```

Comment:
we are providing here, then why singleton in impl class?

Reply:
makes sense. will remove it there...